stock_picking_mgmt_weight/wizards/move_weight.py

```diff
@@ -0,0 +1,137 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import fields, models, api, _
+from datetime import datetime
+from odoo.exceptions import ValidationError
+
+
+class MoveWeight(models.TransientModel):
+    _name = 'stock.move.weight.wizard'
+
+    name = fields.Char('Description')
+    company_id = fields.Many2one('res.company', 'Company',
+                                 default=lambda self: self.env.company,
+                                 index=True,
+                                 required=True)
+    picking_id = fields.Many2one('stock.picking', 'Transfer Reference', index=True, states={'done': [('readonly', True)]}, check_company=True)
+    move_line_ids = fields.One2many(
+        comodel_name='stock.move.line',
+        inverse_name='move_id')
+    move_line_weight_ids = fields.One2many(
+        comodel_name='stock.move.line.weight.wizard',
+        inverse_name='move_weight_id')
+
+    move_id = fields.Many2one('stock.move')
+
+    weight_selection = fields.Selection([
+        ('theoretical_qty', _('Theoretical quantity demanded')),
+        ('net_weight', _('Origin/Destination Weight'))],
+        default='net_weight')
+
+    theoretical_qty = fields.Float(
+        string='Theoretical quantity demanded',
+        digits='Product Unit of Measure')
+    net_weight = fields.Float(
+        string='Net',
+        digits='Product Unit of Measure')
+
+    weight_selected = fields.Float(
+        digits='Product Unit of Measure',
+        compute='_compute_weight_selected')
+    weight_classified = fields.Float(
+        digits='Product Unit of Measure',
+        compute='_compute_weight_classified',
+        readonly=True,
+        store=True)
+
+    @api.onchange('weight_selection')
+    @api.depends('weight_selected')
+    def _compute_weight_selected(self):
+        if self.weight_selection == 'theoretical_qty':
+            self.weight_selected = self.theoretical_qty
+        else:
+            self.weight_selected = self.net_weight
+
+    @api.depends('move_line_weight_ids.product_uom_qty', 'weight_selected')
+    def _compute_weight_classified(self):
+        self.weight_classified = self.weight_selected - sum(self.move_line_weight_ids.mapped('product_uom_qty'))
+
+    def write_operations(self):
+        self.ensure_one()
+        purchase_order = self.env['purchase.order']
+        if self.weight_classified == 0:
+            if not self.move_line_weight_ids.mapped('order_line'):
+
+                order_new = purchase_order.new({
+                    'partner_id': self.picking_id.partner_id.id
+                })
+                order_new.onchange_partner_id()
```

Comment:
Please encapsulate this code in a method, so other addons could set other fields and fire other methods